catalog/spatial/registry/registry-schema-bindings/src/main/java/org/codice/ddf/registry/schemabindings/converter/type/AbstractRegistryObjectTypeConverter.java

```diff
@@ -0,0 +1,180 @@
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.EXTERNAL_IDENTIFIER_KEY;
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.HOME_KEY;
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.ID_KEY;
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.LID_KEY;
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.NAME_KEY;
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.OBJECT_TYPE_KEY;
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.SLOT;
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.STATUS_KEY;
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.VERSION_INFO_KEY;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.codice.ddf.registry.schemabindings.helper.MapToSchemaElement;
+
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ClassificationType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ExternalIdentifierType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryObjectType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.SlotType1;
+
+public abstract class AbstractRegistryObjectTypeConverter<T extends RegistryObjectType> {
+
+    protected abstract MapToSchemaElement<T> getSchemaMapper();
+
+    /**
+     * This method creates a RegistryObjectType from the values in the provided map.
+     * The following keys are expected in the provided map (Taken from EbrimConstants):
+     * <p>
+     * CLASSIFICATION_KEY = "Classification";
+     * EXTERNAL_IDENTIFIER_KEY = "ExternalIdentifier";
+     * NAME_KEY = "Name";
+     * DESCRIPTION_KEY = "Description";
+     * VERSION_INFO_KEY = "VersionInfo";
+     * SLOT = "Slot";
+     * ID_KEY = "id";
+     * HOME_KEY = "home";
+     * LID_KEY = "Lid";
+     * STATUS_KEY = "Status";
+     * OBJECT_TYPE_KEY = "objectType";
+     * <p>
+     * Uses:
+     * ClassificationTypeConverter
+     * ExternalIdentifierTypeConverter
+     * SlotTypeConverter
+     * InternationalStringTypeHelper
+     *
+     * @param map the Map representation of the RegistryObjectType to generate, null returns empty Optional
+     * @return Optional RegistryObjectType created from the values in the map
+     */
+    public Optional<T> convert(Map<String, Object> map) {
+        Optional<T> optionalRegistryObject = Optional.empty();
+
+        if (map.containsKey(CLASSIFICATION_KEY)) {
+            List<Map<String, Object>> classificationMaps = (List<Map<String, Object>>) map.get(
+                    CLASSIFICATION_KEY);
+
+            ClassificationTypeConverter classificationConverter = new ClassificationTypeConverter();
+
+            Optional<ClassificationType> optionalClassification;
```

Comment:
That code (L71-92) is repeated multiple places. We should be able to do add something to the `MapToSchemaElement` class (or create a separate class if we can't) to eliminate all the repetition. Let me know if you want me to help with this.